### ISSUE TITLE
Very low GPU usage during TextClassifier training

### ISSUE THREAD

**Issue by Glorf** [flairNLP/flair#423]
Hello,

I'm using flair (master), pytorch (1.0.0), python3.6 and my GPU (RTX 2060 6GB, CUDA10) on Ubuntu 18.10 x86_64 to develop multilabel classification model for my NLP task on my own corpus.
Most basic parts of my code are as below (following the tutorial):

```
corpus: TaggedCorpus = NLPTaskDataFetcher.load_classification_corpus(data_folder,
                                                                     test_file='test.txt',
                                                                     dev_file='dev.txt',
                                                                     train_file='train.txt').downsample(0.1)
label_dict = corpus.make_label_dictionary()
word_embeddings = [FlairEmbeddings('polish-forward'), FlairEmbeddings('polish-backward')]
document_embeddings: DocumentLSTMEmbeddings = DocumentLSTMEmbeddings(word_embeddings,
                                                                     hidden_size=512,
                                                                     reproject_words=True,
                                                                     reproject_words_dimension=256,
                                                                     )
classifier = TextClassifier(document_embeddings, label_dictionary=label_dict, multi_label=True)
trainer = ModelTrainer(classifier, corpus)
trainer.train('data',
              learning_rate=0.1,
              mini_batch_size=64,
              anneal_factor=0.5,
              patience=5,
              max_epochs=1)
```

While training, my CPU apparently runs one thread, and GPU usage is near-zero. During validation, GPU is apparently used sometimes, but does not run over 25% of cuda cores utilisation.

Please, tell me if I'm doing something wrong or should I check some parameters of my installation. The RAM overflow is not the case (my PC has 12GB, and about 8GB is used during training). GPU RAM overflow also doesn't seem to be the problem, as I get no artifacts nor overflow warnings.
torch.cuda.get_device_name(0) returns my card name properly, so CUDA seems to be configured OK

Thanks in advance for any clues about what happened here

**Comment by Glorf** [comment]
Below is my nvidia-smi output while training - when GPU should be fully used
![nvidia-smi](https://user-images.githubusercontent.com/479124/51759839-6b2ae780-20c9-11e9-9ae0-d623260a82cb.png)

**Comment by alanakbik** [comment]
Thanks for reporting this - we'll take a look!

**Comment by alipetiwala** [comment]
@alanakbik will using CUDA9  instead of CUDA10 help?

**Comment by gccome** [comment]
I recently encountered the same issue. I am running NER with Flair, but GPU utilization never exceeded 25%. Below is my GPU utilization info:
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 396.44                 Driver Version: 396.44                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla K80           On   | 00000000:00:1E.0 Off |                    0 |
| N/A   45C    P0    58W / 149W |   2298MiB / 11441MiB |     17%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0     20229      C   python                                      2287MiB |
+-----------------------------------------------------------------------------+

**Comment by frenzykryger** [comment]
Definitely bug, not question

**Comment by Glorf** [comment]
As it turned out to be reproducible, I change label to `bug`

**Comment by gccome** [comment]
@frenzykryger for fasttext embedding, does Flair support OOV using sub-word?

**Comment by frenzykryger** [comment]
@gccome 
They just call gensim keyedvectors under the hood to get embedding for each token. This should work with OOV:
flair.embeddings.WordEmbeddings#_add_embeddings_internal

**Comment by gccome** [comment]
@frenzykryger please ignore my previous question. I found out that it does support OOV using fasttext embedding.

**Comment by alanakbik** [comment]
Hello @Glorf we just pushed some PRs that fixed some bugs in CPU/GPU processing for text classification. I now get some 30% GPU usage when training text classifiers. Still not great, but better than 1%. 

Could you also test if this works for you?

**Comment by tmills** [comment]
I just noticed this today -- pulled and reinstalled from master and saw an improvement. I noticed because I saw a surprising non-change in speed when increasing the batch size. My own pytorch code seems to reliably run faster with bigger batches, and I'm not close to using up my GPU RAM.

**Comment by gladkikhartem** [comment]
I have the similar situation with RTX2060, pyTorch  and  **autokeras**  (not flair)
I've also tried *Tensoflow* backend and have exactly same GPU usage.
So it looks like this issue is not dependent on the framework.

I agree with @Glorf  that this may be related to CUDA 10 changes and to to specific implementation.
I've used python profiler - most of the time is spend on **backward** **forward** and **math_mul** functions, written in C.

On my machine (6x CPU, 6GB RTX2060, 16GB 3200Mhz memory) I have following behaviour:
CPU usage ~50% on all 12 threads.
GPU usage ~15~25%
RAM usage 14-16 GB for batch size of 32, 10-14 GB for batch size of 24
GPU RAM is not used at all, stays ~ 150MB or so for this process.

Really hope this will be fixed soon, as I've bought whole PC specifically for this task it's now working )))))
unfortunately CUDA 9 is not supported for RTX 20xx series. (((
